packages/paste-website/src/components/icons-list/IconsList.tsx

```diff
@@ -0,0 +1,145 @@
+        return {
+          products: [...prev.products, Icon],
+          ui: [...prev.ui],
+        };
+      }
+      return {
+        products: [...prev.products],
+        ui: [...prev.ui, Icon],
+      };
+    },
+    {products: [], ui: []}
+  );
+
+const getFirstIcon = (iconsList: GroupedList): IconObject | null => {
+  if (iconsList.ui.length !== 0) {
+    return iconsList.ui[0];
+  }
+  if (iconsList.products.length !== 0) {
+    return iconsList.products[0];
+  }
+  return null;
+};
+
+const IconTileWrapper = styled(Box)({
+  width: '33.33%',
+});
+
+const IconColumn = styled(Box)({
+  width: '41.66666666666667%',
+});
+
+const IconsList: React.FC<IconsListProps> = () => {
+  const {isPasteTheme} = React.useContext(SiteBodyContext);
+  const filterID = useUID();
+  const iconKeySeed = useUIDSeed();
+  const groupedIconList = getGroupedList(iconsJson);
+  const [filterString, setFilterString] = React.useState('');
+  const [iconsList, setIconsList] = React.useState(groupedIconList);
+  const [selectedIcon, setSelectedIcon] = React.useState(getFirstIcon(iconsList));
+  return (
+    <Box maxWidth="size120">
+      <Grid gutter="space60">
+        <Column span={7}>
+          <Box marginBottom="space70">
+            <FormLabel htmlFor={filterID}>Filter Icons</FormLabel>
+            <FormInput
+              id={filterID}
+              type="text"
+              value={filterString}
+              placeholder="Filter by name..."
+              onChange={e => {
+                const filteredList = iconsJson.filter(({name}: IconObject) =>
+                  name.toLowerCase().includes(e.currentTarget.value.toLowerCase())
+                );
+                const filteredGroupedList = getGroupedList(filteredList);
+                setIconsList(filteredGroupedList);
+                setSelectedIcon(getFirstIcon(filteredGroupedList));
+                setFilterString(e.currentTarget.value);
+              }}
+            />
+          </Box>
+        </Column>
+        <Column span={7}>
+          {iconsList.ui.length > 0 && (
+            <>
+              <Heading as="h2" variant="heading20">
+                UI Icons
+              </Heading>
+              <Box marginX="spaceNegative40" marginBottom="space70" display="flex" flexWrap="wrap">
+                {iconsList.ui.map((icon, index) => {
+                  if (icon.name !== 'index') {
+                    return (
+                      <IconTileWrapper paddingX="space40" marginBottom="space60" key={iconKeySeed(`icon-${index}`)}>
+                        <IconTile icon={icon} onClick={() => setSelectedIcon(icon)}>
+                          <icon.component decorative size="sizeIcon70" />
+                        </IconTile>
+                      </IconTileWrapper>
+                    );
+                  }
+                  return null;
+                })}
+              </Box>
+            </>
+          )}
+
+          {iconsList.products.length > 0 && (
```

Comment:
Should we hide the `product icon` header and everything if no results are found? Or would it be better to keep the header and show an empty state for the category? 💭

Reply:
Empty state added to the overall page